### PR TITLE
testcase/le_tc/network : add scenario for net_close

### PR DESCRIPTION
**Pull request by arvinmittal**
file path : apps/examples/testcase/le_tc/network
file name : tc_net_close.c

add scenario :
tc_net_close_multi_times_p
tc_net_reclose_n

Signed-off-by: Arvin Mittal <arvin.mittal@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/le_tc/network/Make.defs [hunk 1]
```diff
@@ -54,73 +54,86 @@
 ifeq ($(CONFIG_EXAMPLES_TESTCASE_NETWORK),y)
 CSRCS += network_tc_main.c
 ifeq ($(CONFIG_TC_NET_SOCKET),y)
-CSRCS +=tc_net_socket.c
+CSRCS += tc_net_socket.c
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/Make.defs]: Don't fix different kinds of changes. This is a change of formatting (cosmetic).

File: apps/examples/testcase/le_tc/network/itc_net_close.c [hunk 2]
```diff
@@ -0,0 +1,85 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/itc_net_close.c]: 2017

File: apps/examples/testcase/le_tc/network/itc_net_close.c [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+///file tc_net_socket.c
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/itc_net_close.c]: check comments

File: apps/examples/testcase/le_tc/network/network_tc_main.c [hunk 4]
```diff
@@ -42,7 +42,7 @@ int network_tc_main(int argc, char *argv[])
 	total_pass = 0;
 	total_fail = 0;
 
-	printf("=== TINYARA Network TC START! ===\n");
+	printf("\n########## Network TC Start ##########\n");
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/network_tc_main.c]: Please rebase yours. This was changed already.

File: apps/examples/testcase/le_tc/network/tc_net_close.c [hunk 5]
```diff
@@ -16,59 +16,97 @@
...
-#include <errno.h>
-
-#include <sys/stat.h>
-#include <net/if.h>
-#include <arpa/inet.h>
-#include <netinet/in.h>
-//#include <arch/board/board.h>
-#include <netutils/netlib.h>
-#include <sys/socket.h>
+///file tc_net_socket.c
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/tc_net_close.c]: This file is not matched with your commit message. Please split commits.

File: apps/examples/testcase/le_tc/network/tc_net_close.c [hunk 6]
```diff
@@ -16,59 +16,97 @@
...
+* @apicovered        :close()
+* @precondition      :open socket
+* @postcondition     :none
+*/
 static void tc_net_close_p(void)
 {
-
 	int fd = -1;
 	fd = socket(AF_INET, SOCK_STREAM, 0);
+	TC_ASSERT_GEQ("socket", fd, 0);
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/tc_net_close.c]: If there is a failure on this ASSERT macro, this function will be finished without close(fd). This is a leak.

File: apps/examples/testcase/le_tc/network/tc_net_close.c [hunk 7]
```diff
@@ -77,8 +115,10 @@ static void tc_net_close_n(void)
 
 int net_close_main(void)
 {
-
 	tc_net_close_p();
+	tc_net_close_multi_times_p();
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/network/tc_net_close.c]: This is not an utc. This is an itc. You should name it as itc_xxx. And please put itcs at separated location from utc.
```
tc_xxx1
tc_xxx2

itc_yyy1
itc_yyy2
```

**Comment by sunghan-chang** [comment]
@arvinmittal Additionally, Tizen RT github doesn't need to merge your private commits like "Merge branch 'master' into scenario_net_close". After making P.R, use "git rebase" instead of "git pull". Please remove last two commits.

**Comment by sunghan-chang** [comment]
@arvinmittal These comments are same for another P.R. which your team made.

**Pull request closed without merging**